python/ray/ml/examples/pytorch_geometric/distributed_reddit_example.py

```diff
@@ -0,0 +1,178 @@
+# Adapted from https://github.com/pyg-team/pytorch_geometric/blob/master/examples
+# /multi_gpu/distributed_sampling.py.
+
+import os
+import argparse
+from filelock import FileLock
+
+import torch
+import torch.nn.functional as F
+
+from torch_geometric.datasets import Reddit
+from torch_geometric.loader import NeighborSampler
+from torch_geometric.nn import SAGEConv
+
+from ray import train
+from ray.ml.train.integrations.torch import TorchTrainer
+
+
+class SAGE(torch.nn.Module):
+    def __init__(self, in_channels, hidden_channels, out_channels, num_layers=2):
+        super().__init__()
+        self.num_layers = num_layers
+
+        self.convs = torch.nn.ModuleList()
+        self.convs.append(SAGEConv(in_channels, hidden_channels))
+        for _ in range(self.num_layers - 2):
+            self.convs.append(SAGEConv(hidden_channels, hidden_channels))
+        self.convs.append(SAGEConv(hidden_channels, out_channels))
+
+    def forward(self, x, adjs):
+        for i, (edge_index, _, size) in enumerate(adjs):
+            x_target = x[: size[1]]  # Target nodes are always placed first.
+            x = self.convs[i]((x, x_target), edge_index)
+            if i != self.num_layers - 1:
+                x = F.relu(x)
+                x = F.dropout(x, p=0.5, training=self.training)
+        return x.log_softmax(dim=-1)
+
+    @torch.no_grad()
+    def inference(self, x_all, subgraph_loader):
```

Comment:
Is this planned to be used for predictor impl?

Reply:
Eventually yes, but the challenge for prediction is how to add "fresh data" to the graph to do inference on.